### DISCUSSION TITLE
Warn if there are selections outside of your view

### DISCUSSION THREAD

**Discussion by Ordoviz** [helix-editor/helix#3534] [Ideas] [answered]
A mistake I make frequently with Kakoune and Helix is editing text without realizing that I am using more than one cursor. I wish there was some kind of subtle warning when there are selections outside of your current viewport.

If this is too difficult to implement, then please highlight "N sels" in the statusline in red if N is bigger than 1.

What do you think? Do you also sometimes create multiple selections, forget about them, and start editing – only to realize later that your file has become a total mess?

**Comment by the-mikedavis** [comment]
I remember winding up in this situation a lot in Kakoune. Ideally UI-wise there would be small hints when there are cursors above and/or below the current viewport. Maybe a minimap (https://github.com/helix-editor/helix/issues/2210) would be a good place to show those, or along the top/bottom edges of the view.

In Helix I don't find myself making this mistake though since there are plenty of primitives for selecting specific regions. Tree-sitter motions including textobjects and LSP-driven selections replace `%` for me in most cases and `%` is how I wound up messing up documents in Kakoune.

As a tip: <kbd>(</kbd>/<kbd>)</kbd> are useful tools for figuring out where your selections are when you already know you have many.

**Comment by joegm** [comment, marked as answer]
Duplicate of #2103 ?
